src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DefaultTagHelperCompletionService.cs

```diff
@@ -221,50 +214,13 @@ public override IReadOnlyList<CompletionItem> GetCompletionsAt(SourceSpan locati
             return completionItems;
         }
 
-        private bool TryResolveAttributeInsertionSnippet(
-            string text,
-            IEnumerable<BoundAttributeDescriptor> boundAttributes,
-            bool indexerCompletion,
-            out string snippetText)
-        {
-            var snippetSupported = LanguageServer.ClientSettings?.Capabilities?.TextDocument?.Completion.Value?.CompletionItem?.SnippetSupport ?? false;
-            if (!snippetSupported)
-            {
-                snippetText = null;
-                return false;
-            }
-
-            const string BoolTypeName = "System.Boolean";
-
-            // Boolean returning bound attribute, auto-complete to just the attribute name.
-            if (indexerCompletion)
-            {
-                if (boundAttributes.All(boundAttribute => boundAttribute.IndexerTypeName == BoolTypeName))
-                {
-                    snippetText = null;
-                    return false;
-                }
-
-                snippetText = string.Concat(text, "$1=\"$2\"");
-                return true;
-            }
-            else if (boundAttributes.All(boundAttribute => boundAttribute.TypeName == BoolTypeName))
-            {
-                snippetText = null;
-                return false;
-            }
-
-            snippetText = string.Concat(text, "=\"$1\"");
-            return true;
-        }
-
         private Container<string> ResolveAttributeCommitCharacters(IEnumerable<BoundAttributeDescriptor> boundAttributes, bool indexerCompletion)
         {
             if (indexerCompletion)
             {
                 return NoCommitCharacters;
             }
-            else if (boundAttributes.Any(b => b.TypeName.StartsWith("System.Boolean")))
+            else if (boundAttributes.Any(b => b.TypeName == "System.Boolean"))
```

Comment:
This was meant for properties of type `System.Boolean` however such types need to be explicitly specified with a `true/false` and whitespace isn't sufficient.